src/js/components/ClinPhen/TimeElementDisplay.tsx

```diff
@@ -0,0 +1,104 @@
+import { memo } from 'react';
+
+import { EM_DASH } from '@/constants/contentConstants';
+import type {
+  TimeElement,
+  TimeElementAge,
+  TimeElementAgeRange,
+  TimeElementGestationalAge,
+  TimeElementInterval,
+  TimeElementOntologyClass,
+  TimeElementTimestamp,
+  TimeInterval,
+} from '@/types/clinphen/shared';
+
+type TimeElementType = 'age' | 'gestational_age' | 'age_range' | 'ontology_class' | 'timestamp' | 'interval';
+
+const TIME_ELEMENT_TYPES_LABELS: Record<TimeElementType, string> = {
+  age: 'Age',
+  gestational_age: 'Gestational Age',
+  age_range: 'Age Range',
+  ontology_class: 'Ontology Class',
+  timestamp: 'Timestamp',
+  interval: 'Interval',
+};
+
+const getTimeElementTypeLabel = (element: TimeElement): [TimeElementType | null, string] => {
+  const keys = Object.keys(element);
+  if (keys.length === 1) {
+    // A Phenopacket TimeElement should only have 1 property
+    const type = keys[0] as TimeElementType;
+    if (type in TIME_ELEMENT_TYPES_LABELS) {
+      const label = TIME_ELEMENT_TYPES_LABELS[type];
+      return [type, label];
+    }
+  }
+  return [null, 'NOT_SUPPORTED'];
+};
+
+export const TimeIntervalDisplay = ({ timeInterval, br }: { timeInterval: TimeInterval; br?: boolean }) => {
```

Comment:
Wanted to know why there is an inconsistency of some return components being wrapped in <span> while others not?

Reply:
these are all ported from Bento Web - i think this is used elsewhere originally which is why it's exported. i will check.

Reply:
specifically, it's used for displaying medical actions https://github.com/bento-platform/bento_web/blob/f2f92657fc4293e35ac4351eeb83855e6d534790/src/components/explorer/IndividualMedicalActions.js#L69